theia-extensions/viewer-prototype/src/browser/trace-explorer/trace-explorer-sub-widgets/trace-explorer-placeholder-widget.tsx

```diff
@@ -94,19 +94,22 @@ export class TraceExplorerPlaceholderWidget extends ReactWidget {
                 text: ''
             }).then(async progress => {
                 progress.report({ message: 'Launching trace server... ', work: { done: 10, total: 100 } });
+                const { path, args } = this;
                 try {
-                    const resolve = await this.traceServerConfigService.startTraceServer(this.path, this.port);
+                    const resolve = await this.traceServerConfigService.startTraceServer({ path, args });
                     if (resolve === 'success') {
-                        progress.report({ message: `Trace server started on port: ${this.port}.`, work: { done: 100, total: 100 } });
+                        progress.report({ message: `Trace server started using the following arguments:  ${this.args}.`, work: { done: 100, total: 100 } });
                         progress.cancel();
                         this.commandService.executeCommand(OpenTraceCommand.id);
                     }
                 }
                 catch (err) {
                     // eslint-disable-next-line @typescript-eslint/no-explicit-any
                     if (PortBusy.is(err as any)) {
+                        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                        const port = this.args!.match(/(?<=port=)\d+/g)![0];
                         this.messageService.error(
-                            `Error opening serial port ${this.port}. (Port busy)`);
+                            `Error opening serial port ${port}. (Port busy)`);
```

Comment:
Extracting the port parameter is server-specific. The message can just output the whole args string:
`Error starting the server (port busy) using the following arguments: ${this.args}`